.github/workflows/c-cpp.yml

```diff
@@ -0,0 +1,60 @@
+name: C/C++ CI
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        config: [Release]
+
+    steps:
+      - name: Set up build environment (macos-latest)
+        run: |
+          brew install ccache boost
```

Comment:
Should we use pre-built `boost`? Someone could break local `boost` or use some `boost` features that haven't beed added to `ext-boost` and CI will ignore it. 🙈

Reply:
Or should we have builds with pre-built `boost` and others without to cover all cases? Currently, I think only the macOS builds on Travis use `boost` from the submodule.

Reply:
What should we do about pre-built binaries @1whatleytay ?